mono/metadata/object.c

```diff
@@ -6998,7 +6998,7 @@ mono_object_get_size (MonoObject* o)
 
 	MonoClass* klass = mono_object_class (o);
 	if (klass == mono_defaults.string_class) {
-		return sizeof (MonoString) + 2 * mono_string_length ((MonoString*) o) + 2;
+		return MONO_SIZEOF_MONO_STRING + 2 * mono_string_length ((MonoString*) o) + 2;
```

Comment:
To illustrate the issue, see: https://gcc.godbolt.org/z/g1rhHc

It wasn't obvious to me that `sizeof` _includes_ the empty `chars` array. Good catch.